server/src/main/java/org/opensearch/search/internal/InternalSearchResponse.java

```diff
@@ -80,6 +97,8 @@ public InternalSearchResponse(StreamInput in) throws IOException {
             in.readOptionalWriteable(SearchProfileShardResults::new),
             in.readVInt()
         );
+
+        this.searchExtBuilders.addAll(readSearchExtBuildersOnOrAfter(in, Version.V_3_0_0));
```

Comment:
Super nit, I think you could safely hardcode the version parameter in the method, there is literally no way it would change once we get the backport completed.